api/app/controllers/spree/api/v2/storefront/taxons_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class TaxonsController < ::Spree::Api::V2::BaseController
+          def index
+            render json: serialize_taxons_collection(collection), status: 200
+          end
+
+          def show
+            render json: serialize_taxon_resource(resource), status: 200
+          end
+
+          private
+
+          def dependencies
+            {
+              taxons_finder:                Spree::Taxons::Find,
+              taxon_resource_serializer:    Spree::V2::Storefront::TaxonSerializer,
+              taxons_collection_serializer: Spree::V2::Storefront::TaxonSerializer
+            }
+          end
+
+          def collection
+            @taxons = Spree::Taxon.includes(:parent, :children).accessible_by(current_ability, :read)
```

Comment:
@bbonislawski 
How about extracting `includes` to a separate method so that it's easy to override for other developers?

Reply:
I'm worried that it can be a little too much and I'd suggest to override collection method if someone wants to change it somehow. Let's discuss it today in office